File: backend/src/main/java/com/Fijalkowskim/SpaceFarmManagmentSystem/requestmodels/CultivationRequest.java
package com.Fijalkowskim.SpaceFarmManagmentSystem.requestmodels;

import com.Fijalkowskim.SpaceFarmManagmentSystem.models.*;
import com.Fijalkowskim.SpaceFarmManagmentSystem.models.dictionaries.CultivationType;
import lombok.Data;

import java.util.Date;
import java.util.Set;

@Data
public class CultivationRequest {
    private Date startDate;
    private CultivationType type;
    private float area;
    private Date plannedFinishDate;
    private Date realFinishDate;
    private Plant plant;
    private String comment;
}

File: frontend/src/models/requestmodels/CultivationRequest.js
export class CultivationRequest {
  constructor(
    startDate = new Date(),
    plannedFinishDate = undefined,
    realFinishDate = undefined,
    plant = undefined,
    comment = "",
    area = 0,
    type = undefined
  ) {
    this.startDate = startDate;
    this.plannedFinishDate = plannedFinishDate;
    this.realFinishDate = realFinishDate;
    this.plant = plant;
    this.comment = comment;
    this.area = area;
    this.type = type;
  }
}
